PyInstaller/building/splash.py

```diff
@@ -0,0 +1,308 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2019-2020, PyInstaller Development Team.
+#
+# Distributed under the terms of the GNU General Public License (version 2
+# or later) with exception for distributing the bootloader.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#
+# SPDX-License-Identifier: (GPL-2.0-or-later WITH Bootloader-exception)
+# -----------------------------------------------------------------------------
+
+import array
+import os
+import struct
+import sys
+
+from .. import log as logging
+from .datastruct import Target
+from .utils import misc, _check_guts_eq
+from ..compat import is_win
+from ..utils.misc import Structure, grouper
+
+logger = logging.getLogger(__name__)
+
+
+class BITMAPFILEHEADER(Structure):
+    """BITMAPFILEHEADER is at the start of any Windows bitmap file.
+
+    All of the integer values are stored in little-endian format
+    """
+    _names_ = ("bfType", "bfSize", "bfReserved1", "bfReserved2", "bfOffBits")
+    _format_ = "<2s L H H L"
+
+
+class BITMAPINFOHEADER(Structure):
+    """BITMAPINFOHEADER stores all the image-related data for an bitmap.
+
+    All of the integer values are stored in little-endian format.
+    """
+    _names_ = ("biSize", "biWidth", "biHeight", "biPlanes", "biBitCount",
+               "biCompression", "biSizeImage", "biXPelsPerMeter",
+               "biYPelsPerMeter", "biClrUsed", "biClrImportant")
+    _format_ = "<LllHHLLllLL"
+
+
+def _convert_hex_to_le(rgb):
+    """
+    Converts a RGB value specified by 0xRRGGBB into 0xBBGGRR.
+    """
+    if not 0x00 <= rgb <= 0xFFFFFF:
```

Comment:
This should be extracted into `Splash`, because the rgb color in hex is always between `0x00` and `0xFFFFFF`, not only on Windows